check-github-issue.py

```diff
@@ -0,0 +1,116 @@
+    # TODO: check if we got valid data before returning the same?
+    return parsed_output.get("message")
+
+
+def parse_commit_message(msg):
+    regex = re.compile(r'(([fF][iI][xX][eE][sS])|([uU][pP][dD][aA][tT][eE][sS])):?\s+(gluster/glusterfs)?#(\d*)')
+    issues = []
+    for line in msg.split('\n'):
+        for match in regex.finditer(line):
+            issue=match.group(5)
+            if issue.to_i >= 743000:
+                continue
+            issues.append(unicode(issue))
+    if len(issues):
+        print("Issues found in the commit message: {}".format(issues))
+        return issues
+    print("No issues found in the commit message")
+    return issues
+
+
+def remove_duplicates (project, branch, change_id, revision_number, issues):
+    if (int(revision_number) == 1):
+        return issues
+
+    oldmessage = get_commit_message_from_gerrit(project, branch, change_id,
+                                                str(int(revision_number) - 1))
+    oldissues = parse_commit_message(oldmessage)
+    print("Old issues: {}".format(oldissues))
+    newissues = list(set(issues) - (set(issues) & set(oldissues)))
+    return newissues
+
+def check_issue(num, comment, dry_run):
+    if dry_run:
+        print comment
+    else:
+        gh_user = os.environ.get('GITHUB_USER')
+        gh_pw = os.environ.get('GITHUB_PASS')
+        gh = login(gh_user, gh_pw)
+        issue = gh.issue('gluster', 'glusterfs', num)
+        if issue:
+            spec_approved = False
+            doc_approved = False
+            for lbl in issue.labels:
+                if lbl == "SpecApproved":
+                    spec_approved = True
+                if lbl == "DocApproved":
+                    doc_approved = True
+            if spec_approved and doc_approved:
+                print "All approvals in place"
+                exit (0)
+            else:
+                print "Missing the required approvals"
+                exit (1)
+        else:
+            print "Issue #{} does not exist".format(num)
+            print "FAIL"
+            exit(1)
+
+def main(dry_run):
+    project = os.environ.get('GERRIT_PROJECT')
+    branch = os.environ.get('GERRIT_BRANCH')
+    change_id = os.environ.get('GERRIT_CHANGE_ID')
+    revision_number = os.environ.get('GERRIT_PATCHSET_NUMBER')
+    link = os.environ.get('GERRIT_CHANGE_URL')
+
+    # get commit message
+    commit = get_commit_message()
+
+    # check if updates/fixes: xxx appears in the commit
+    issues = parse_commit_message(commit)
+
+    if issues:
+        # remove duplicates from previous commit message (if any)
+        newissues = remove_duplicates(project, branch, change_id, revision_number,
+                                      issues)
+        for issue in newissues:
+            # comment on issue: xxx about the review
+            check_issue(issue, commit, dry_run)
+
+
+parser = argparse.ArgumentParser(description='Comment on Github issue')
```

Comment:
Assuming comments here and above would be refreshed to point to this scripts workings.

(nit) On further thought, should we build a library that both update-issue.py and this one uses, that way fixes in the future are in one copy?

Reply:
that would be great.